app/Http/Utilities/Email.php

```diff
@@ -2,17 +2,120 @@
 
 namespace Gladiator\Http\Utilities;
 
+use Gladiator\Models\Contest;
+use Gladiator\Models\Competition;
+use Gladiator\Models\Message;
+
 class Email
 {
-    public static function prepareMessage($message, $model)
+    /**
+     * Message instance.
+     *
+     * @var \Gladiator\Models\Message
+     */
+    protected $message;
+
+    /**
+     * Competition instance.
+     *
+     * @var \Gladiator\Models\Competition
+     */
+    protected $competition;
+
+    /**
+     * Northstar user
+     *
+     * @var object
+     */
+    protected $users;
+
+    /**
+     * Contest instance.
+     *
+     * @var \Gladiator\Models\Contest
+     */
+    public $contest;
+
+    /**
+     * Array of all processed messages for per user
+     *
+     * @var array
+     */
+    public $allMessages;
+
+    /**
+     * Constructor
+     */
+    public function __construct(Message $message, Contest $contest, Competition $competition, $users)
     {
-        //@TODO: I only tested this with competition, is there other things we want to replace?
-        $tokens = $model::$tokenizable;
-        foreach ($tokens as $token) {
-            $message->body = str_replace(':'.$token, $model->$token, $message->body);
-                //@TODO: handle the message subject.
-        }
+        $this->message = $message;
+        $this->contest = $contest;
+        $this->competition = $competition;
+        $this->users = $users;
+
+        $this->setupEmail();
+    }
+
+    /**
+     * Process all of the tokens for this email. New tokens should be defined here.
+     * @TODO - possibly move token stuff into it's own class.
+     *
+     * @param  object $user
+     * @return array $tokens
+     */
+    protected function defineTokens($user)
+    {
+        // @TODO - set defaults?
+        $tokens = [
+            ':competition_end_date' => $this->competition->competition_end_date,
+            ':leaderboard_msg_day' => $this->competition->leaderboard_msg_day,
```

Comment:
ah yea...whatever you think is best! i was thinking there would be another issue for when we need to add more tokens. I thought #153 was just for discussion since it was just labeled as a question